File: app/src/main/java/com/wkz/smartprogressbar/MainActivity.java
package com.wkz.smartprogressbar;

import android.os.Bundle;

import androidx.appcompat.app.AppCompatActivity;

public class MainActivity extends AppCompatActivity {

    private SmartProgressBar mSpbHorizontal;
    private SmartProgressBar mSpbVertical;
    private SmartProgressBar mSpbRing1;
    private SmartProgressBar mSpbRing2;
    private SmartProgressBar mSpbSector1;
    private SmartProgressBar mSpbSector2;
    private ProgressBarLayout mPblProgress;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        initView();
        initData();
    }

    private void initView() {
        mSpbHorizontal = (SmartProgressBar) findViewById(R.id.spb_horizontal);
        mSpbVertical = (SmartProgressBar) findViewById(R.id.spb_vertical);
        mSpbRing1 = (SmartProgressBar) findViewById(R.id.spb_ring_1);
        mSpbRing2 = (SmartProgressBar) findViewById(R.id.spb_ring_2);
        mSpbSector1 = (SmartProgressBar) findViewById(R.id.spb_sector_1);
        mSpbSector2 = (SmartProgressBar) findViewById(R.id.spb_sector_2);
        mPblProgress = (ProgressBarLayout) findViewById(R.id.pbl_progress);
    }

    private void initData() {
    }
}

File: app/src/main/java/com/wkz/smartprogressbar/MainActivity.kt
package com.wkz.smartprogressbar

import android.animation.Animator
import android.animation.AnimatorListenerAdapter
import android.os.Bundle
import android.util.Log
import androidx.appcompat.app.AppCompatActivity
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        initView()
        initListener()
    }

    private fun initView() {
        pblProgress.mSmartProgressBar.setIsAnimated(false)
        val beginTemperature = 90f
        val endTemperature = 140f
        pblProgress.max = 10f
        pblProgress.setTemperatureText(beginTemperature, endTemperature)
        pblProgress.setProgressAnimatorListener(object : AnimatorListenerAdapter() {
            override fun onAnimationEnd(animation: Animator?) {
                super.onAnimationEnd(animation)
                Log.i("onAnimationEnd", "onAnimationEnd")
            }
        })
    }

    private fun initListener() {
        btnPause.setOnClickListener {
            pblProgress.pauseProgressAnimation()
        }
        btnResume.setOnClickListener {
            pblProgress.resumeProgressAnimation()
        }
        btnCancel.setOnClickListener {
            pblProgress.cancelProgressAnimation()
        }
    }
}